### PR TITLE
adding config option to load secrets from a file rather than a hardco…

### PR DESCRIPTION
**Pull request by kris-watts-gravwell**
…ded string

<!--

### CODE REVIEW

File: ingest/config/env.go [hunk 1]
```diff
@@ -491,3 +496,42 @@ func loadEnvVarList(lst *[]string, envName string) error {
...
+
+	//if we hit here, it means we have a size that is ok, load a buffer and read it
+	buff := make([]byte, sz)
+	_, err = io.ReadFull(fin, buff) //attempt the read
+	fin.Close()                     //close the file, because we don't need it anymore
+	if err != nil {
+		return fmt.Errorf("failed to read complete string from %q %w", pth, err)
+	}
+	//trim nulls, newlines, and carriage returns from the file because it will be a nightmare to debug
+	*val = string(bytes.Trim(buff, "\n\t\r\x00"))
```
Comment (david-fritz-gravwell) [inline ingest/config/env.go:534]: Does TrimSpace not do this in a more idiomatic way? Or is this to cover the null character case?

**Review by david-fritz-gravwell — CHANGES_REQUESTED** [review]

File: ingest/config/env.go (hunk 1 above)
Comment (kris-watts-gravwell) [reply ingest/config/env.go:534]: and i don't want to trim spaces, because we have a few people using them in their secrets... :/

**Comment by kris-watts-gravwell** [comment]
@david-fritz-gravwell PTAL

**Review by david-fritz-gravwell — APPROVED** [review]
LGTM

**Pull request merged**